source/Halibut.Tests/Transport/RewindableBufferStreamTests.cs

```diff
@@ -44,6 +45,25 @@ public async Task ReadAsyncShouldPassThrough()
                 }
             }
         }
+        
+        [Test]
+        public async Task ReadAsyncShouldPassTheCancellationTokenToTheUnderlyingStream()
+        {
+            using (var baseStream = new MemoryStream(16))
+            using (var sut = RewindableBufferStreamBuilder.Build(baseStream))
+            {
+                var inputBuffer = Encoding.ASCII.GetBytes("Test");
+                await baseStream.WriteAsync(inputBuffer, 0, inputBuffer.Length);
+
+                baseStream.Position = 0;
+
+                var readBuffer = new byte[inputBuffer.Length];
+                var cts = new CancellationTokenSource();
+                cts.Cancel();
+                
+                Assert.ThrowsAsync<TaskCanceledException>(async () => await sut.ReadAsync(readBuffer, 0, readBuffer.Length, cts.Token));
```

Comment:
Amazingly this is not something that is awaited!

Reply:
Looks like we are already using FluentAssertions in the tests so you could use `Should().ThrowAsync<TaskCanceledException>` if you wanted to.

Reply:
oh let me try that.

Reply:
oh nice that worked.